client/src/SessionController.js

```diff
@@ -116,70 +117,71 @@ export function SessionController({ children }) {
 
   useEffect(() => {
     if (userStates.status === REQUEST_STATE.LOGGINGIN) {
+      dispatch(fetchLimitLeftAsync())
       delaySetLoadingFalse(2500, () => dispatch(updateAsLoggedIn()))
     } else if (userStates.status === REQUEST_STATE.REJECTED) {
       delaySetLoadingFalse(2500, () => dispatch(updateAsLoggedOut()))
     }
-  }, [userStates, dispatch])
+  }, [userStates.status, dispatch])
 
   useEffect(() => {
     if (tripsStates.error) {
-      setAlertMessage(tripsStates.error)
-      setAlertOpen(true)
-      setIsLoading(false)
+      enqueueSnackbar(tripsStates.error, { variant: 'error' })
       dispatch(clearTripsError())
       dispatch(openSidebar())
     }
-  }, [dispatch, tripsStates.error])
+  }, [dispatch, enqueueSnackbar, tripsStates.error])
 
   useEffect(() => {
     if (stagesStates.error) {
-      setAlertMessage(stagesStates.error)
-      setAlertOpen(true)
-      setIsLoading(false)
+      enqueueSnackbar(stagesStates.error, { variant: 'error' })
       dispatch(clearStagesError())
       dispatch(openSidebar())
     }
-  }, [dispatch, stagesStates.error])
+  }, [dispatch, enqueueSnackbar, stagesStates.error])
 
   useEffect(() => {
     if (userStates.error) {
       setTimeout(() => {
-        setAlertMessage('Error: Invalid Credentials')
-        setAlertOpen(true)
+        enqueueSnackbar('Error: Invalid Credentials', { variant: 'error' })
         dispatch(clearUserError())
-        setIsLoading(false)
       }, 2500)
     }
-  }, [dispatch, userStates.error])
+  }, [dispatch, enqueueSnackbar, userStates.error])
 
-  const handleCloseAlert = (event, reason) => {
-    if (reason === 'clickaway') {
-      return
+  useEffect(() => {
+    if (
+      modalStates.newTripModalIsOpen ||
+      modalStates.editStageModalIsOpen ||
+      modalStates.editTripModalIsOpen
+    ) {
+      enqueueSnackbar(
+        `You have ${userStates.attemptLeft} trip creation/update requests left today`,
+        { variant: userStates.attemptLeft > 3 ? 'info' : 'warning' },
+      )
     }
-    setAlertOpen(false)
-  }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [modalStates, enqueueSnackbar])
 
   useEffect(() => {
-    if (alertOpen) {
-      const timer = setTimeout(() => {
-        setAlertOpen(false)
-      }, 5000)
-      return () => clearTimeout(timer)
+    if (userStates.status === REQUEST_STATE.LOGGEDIN) {
```

Comment:
opinion: i don't think the user needs to know how many trips they have left unless they're trying to edit smt. but if u want to leave this in, it's fine by me 🏕️

Reply:
o nvm it's by IP. then this is ok